### ISSUE TITLE
BUG : Funky behaviour in photo editor (of GridField ?)

### ISSUE THREAD

**Issue by SpiritLevel** [silverstripe/silverstripe-framework#5134]
Just noticed this: the thumbnail and the version presented when Edit clicked are different ! I am using a GridField with GridFieldConfig_RecordEditor.

![screenshot from 2016-03-02 21 33 29](https://cloud.githubusercontent.com/assets/8912185/13484878/0dc6e028-e0bf-11e5-952d-d6daf8efef12.png)

**Comment by SpiritLevel** [comment]
It corrects itself upon page refresh..don't have time to look into this so just noting it here. I've also noticed strange behaviour with file sync.

**Comment by SpiritLevel** [comment]
Sorry....periods in file names in addition to the one separating basename from extension seems to be the culprit. Closing.

**Comment by jonom** [comment]
@spiritlevel no need to apologize - extra periods in file names are valid so if that is causing this bug I think we should fix it. Are you able to provide steps to reproduce, including version numbers?

**Comment by SpiritLevel** [comment]
Hey @jonom .... Oh, I thought I saw a conversation about how extra periods in filenames were not allowed...and it seemed my problem went away when I eliminated them! Also, I had a period in the folder name (see above screenshot "St."). I'll try to reproduce. The above screenshot was in version 3.3. 

File syncing did a few odd things too. One was that it wouldn't detect a folder that was in assets. Another is that it deleted a non-empty folder in assets, which is a bit worrisome. Also, in the GridField CMS table, I sometimes wasn't able to delete photos. To get past this strange behaviour, I started from scratch by manually removing the folders and photos from assets, re-syncing files in the CMS, and re-uploading the photos.
